### ISSUE TITLE
Empty "tuple variant defined here" note

### ISSUE THREAD

**Issue by jruderman** [rust-lang/rust#105241]
### Code

[playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=08b23c1f9b05ebc97c3a763e6d4df851)

```rust
fn f() -> Option<()> {
    Some(0_u32)
}

fn main() {}
```

### Current output

```
error[E0308]: mismatched types
 --> tvdh.rs:2:10
  |
2 |     Some(0_u32)
  |     ---- ^^^^^ expected `()`, found `u32`
  |     |
  |     arguments to this enum variant are incorrect
  |
note: tuple variant defined here

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
```

The note starting with "note: tuple variant defined here" is confusing. Where is "here"?

I suspect it's trying to to show part of the definition of the `Option` type, but failing because the definition is in a different crate (or because it's part of `std`).

### Ideal output

Don't try to show the enum-variant definition when it can't be shown, I guess?

For cases like this one that involve generics, see #105240 for a proposed alternative.

### Regression

The empty note first appears in nightly-2022-06-12

### Version

```
rustc 1.67.0-nightly (32e613bba 2022-12-02)
binary: rustc
commit-hash: 32e613bbaafee1bcabba48a2257b838f8d1c03d3
commit-date: 2022-12-02
host: x86_64-apple-darwin
release: 1.67.0-nightly
LLVM version: 15.0.4
```

**Comment by jieyouxu** [comment]
This seems to have been fixed, on stable (1.69.0) it produces

```rust
error[E0308]: mismatched types
   --> test.rs:2:10
    |
2   |     Some(0_u32)
    |     ---- ^^^^^ expected `()`, found `u32`
    |     |
    |     arguments to this enum variant are incorrect
    |
help: the type constructed contains `u32` due to the type of the argument passed
   --> test.rs:2:5
    |
2   |     Some(0_u32)
    |     ^^^^^-----^
    |          |
    |          this argument influences the type of `Some`
note: tuple variant defined here
   --> /home/jieyouxu/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:571:5
    |
571 |     Some(#[stable(feature = "rust1", since = "1.0.0")] T),
    |    
```

```
rustc 1.69.0 (84c898d65 2023-04-16)
binary: rustc
commit-hash: 84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc
commit-date: 2023-04-16
host: x86_64-unknown-linux-gnu
release: 1.69.0
LLVM version: 15.0.7
```

**Comment by jruderman** [comment]
The filename option.rs is now shown (fixed in #104449)

The new hint "... due to the type of the argument passed" was added in #106752
